consensus/spos/bls/blsSubroundsFactory.go

```diff
@@ -31,9 +35,10 @@ func NewSubroundsFactory(
 	}
 
 	fct := factory{
-		consensusCore:  consensusDataContainer,
-		consensusState: consensusState,
-		worker:         worker,
+		consensusCore:    consensusDataContainer,
+		consensusState:   consensusState,
+		worker:           worker,
+		appStatusHandler: statusHandler.NewNilStatusHandler(),
```

Comment:
Because as we decided when implementing AppStatusHandler, it is not a too important component so it didn't make sense to pass it through constructors, check for nil all the time, change testes, etc. Instead, we used setters and with their help we can give a real implementation such as prometheus, termui, etc. So instead of checking for nils all the time when calling a method from appstatushandler (which would be time consuming) we set up an instance of NilStatusHandler which doesn't do anything, but this way the status handler will be nil pointer safe.  If some flags and conditions would be set one ore more real implementations would be set

Reply:
o...k...ish..